plasmapy/dispersion/two_fluid_dispersion.py

```diff
@@ -0,0 +1,352 @@
+    of Bellan 2012 [1]_):
+
+    .. math::
+        \frac{\omega}{\omega_{ci}} = \sqrt{
+            2 \Lambda \sqrt{-\frac{P}{3}} \cos\left(
+                \frac{1}{3} \cos^{-1}\left(
+                    \frac{3q}{2p} \sqrt{-\frac{3}{p}}
+                \right)
+                - \frac{2 \pi}{3}j
+            \right)
+            + \frac{\Lambda A}{3}
+        }
+
+    where :math:`j = 0` represents the fast mode, :math:`j = 1` represents the
+    Alfvén mode, and :math:`j = 2` represents the Acoustic mode.  Additionally,
+
+    .. math::
+        p &= \frac{3B-A^2}{3} \; , \; q = \frac{9AB-2A^3-27C}{27} \\
+        A &= \frac{Q + Q^2 \beta + Q \alpha + \alpha \Lambda}{Q^2} \;
+            , \; B = \alpha \frac{1 + 2 Q \beta + \Lambda \beta}{Q^2} \;
+            , \; C = \frac{\alpha^2 \beta}{Q^2} \\
+        \alpha &= \cos^2 \theta \;
+            , \; \beta = \left( \frac{c_s}{v_A}\right)^2 \;
+            , \; \Lambda = \left( \frac{k v_{A}}{\omega_{ci}}\right)^2
+
+    References
+    ----------
+    .. [1] PM bellan, Improved basis set for low frequency plasma waves, 2012,
+       JGR, 117, A12219, doi: `10.1029/2012JA017856
+       <https://agupubs.onlinelibrary.wiley.com/doi/10.1029/2012JA017856>`_.
+
+    .. [2] TE Stringer, Low-frequency waves in an unbounded plasma, 1963, JNE,
+       Part C, doi: `10.1088/0368-3281/5/2/304
+       <https://doi.org/10.1088/0368-3281/5/2/304>`_
+
+    Examples
+    --------
+    >>> from astropy import units as u
+    >>> from plasmapy.dispersion import two_fluid_dispersion
+    >>> inputs = {
+    ...     "k": 0.01 * u.rad / u.m,
+    ...     "theta": 30 * u.deg,
+    ...     "B": 8.3e-9 * u.T,
+    ...     "n_i": 5e6 * u.m ** -3,
+    ...     "T_e": 1.6e6 * u.K,
+    ...     "T_i": 4.0e5 * u.K,
+    ...     "ion": "p+",
+    ... }
+    >>> omegas = two_fluid_dispersion_solution(**inputs)
+    >>> omegas
+    {'fast_mode': <Quantity 1520.57... rad / s>,
+     'alfven_mode': <Quantity 1261.75... rad / s>,
+     'acoustic_mode': <Quantity 0.688152... rad / s>}
+
+    >>> inputs = {
+    ...     "k": [1e-7, 2e-7] * u.rad / u.m,
+    ...     "theta": [10, 20] * u.deg,
+    ...     "B": 8.3e-9 * u.T,
+    ...     "n_i": 5e6 * u.m ** -3,
+    ...     "T_e": 1.6e6 * u.K,
+    ...     "T_i": 4.0e5 * u.K,
+    ...     "ion": "He+",
+    ... }
+    >>> omegas = two_fluid_dispersion_solution(**inputs)
+    >>> omegas['fast_mode']
+    <Quantity [[0.00767..., 0.00779... ],
+               [0.01534..., 0.01558...]] rad / s>
+    """
+
+    # validate argument ion
+    if not isinstance(ion, Particle):
+        try:
+            ion = Particle(ion)
+        except TypeError:
+            raise TypeError(
+                f"For argument 'ion' expected type {Particle} but got {type(ion)}."
+            )
+    if not (ion.is_ion or ion.is_category("element")):
+        raise ValueError(f"The particle passed for 'ion' must be an ion or element.")
```

Comment:
If you don't mind, I'd like to add `@particle_input` after we develop and deploy `particle_like`.  `@particle_input` just doesn't play well with other decorators so I'd like to avoid unforeseen complications until `@particle_input` goes through its update.